core/src/syn/v2/parser/prime.rs

```diff
@@ -456,11 +515,45 @@ impl Parser<'_> {
 			}
 		};
 		if let Some(start) = start {
+			if self.peek_kind() != t!(")") && Self::starts_disallowed_subquery_statement(peek.kind)
+			{
+				if let Subquery::Value(Value::Idiom(Idiom(ref idiom))) = res {
+					if idiom.len() == 1 {
+						// we parsed a single idiom and the next token was a dissallowed statement so
+						// it is likely that the used meant to use an invalid statement.
+						return Err(ParseError::new(
+							ParseErrorKind::DisallowedStatement {
+								found: self.peek_kind(),
+								expected: t!(")"),
+								disallowed: peek.span,
+							},
+							self.recent_span(),
+						));
+					}
+				}
+			}
+
 			self.expect_closing_delimiter(t!(")"), start)?;
 		}
 		Ok(res)
 	}
 
+	fn starts_disallowed_subquery_statement(kind: TokenKind) -> bool {
+		matches!(
+			kind,
+			t!("ANALYZE")
+				| t!("BEGIN") | t!("BREAK")
+				| t!("CANCEL") | t!("COMMIT")
+				| t!("CONTINUE") | t!("FOR")
+				| t!("INFO") | t!("INSERT")
+				| t!("KILL") | t!("LIVE")
+				| t!("OPTION") | t!("RELATE")
+				| t!("LET") | t!("SHOW")
+				| t!("SLEEP") | t!("THROW")
+				| t!("USE")
```

Comment:
CONTINUE, FOR, INSERT, RELATE, LET, THROW, and possibly SLEEP, KILL, and LIVE are allowed as subqueries AFAIK

Reply:
Ah wait, in which context is this actually?

Reply:
These are not allowed in a subquery but they are allowed inside a block statement.

You can't do `1 + (CONTINUE)` but you can do `1 + { CONTINUE; }` I guess.

Reply:
Gotcha! I think you can do some of the statements though

Reply:
No? You can even look at the old parser, the only statements allowed in a subquery are `RETURN` `SELECT` `CREATE` `UPDATE` `DELETE` `RELATE` `INSERT` `DEFINE` `REMOVE` and `IF`.